dali/pipeline/data/buffer.h

```diff
@@ -307,7 +326,45 @@ class DLL_PUBLIC Buffer {
     return shares_data_;
   }
 
-  DISABLE_COPY_MOVE_ASSIGN(Buffer);
+  /**
+   * @brief Set another Buffer as the backing memory for this Buffer.
+   *
+   * Current Buffer will be marked as sharing data, and reallocation of memory will be
+   * prohibited until reset() is called.
+   */
+  inline void set_backing_allocation(const Buffer<Backend> &other) {
+    type_ = other.type_;
+    data_ = other.data_;
+    allocate_ = {};
+    size_ = other.size_;
+    shares_data_ = true;
+    num_bytes_ = other.num_bytes_;
+    device_ = other.device_;
+  }
+
+  /**
+   * @brief Set external memory as the backing memory for this Buffer.
+   *
+   * Current Buffer will be marked as sharing data, and reallocation of memory will be
+   * prohibited until reset() is called.
+   *
+   * For GPU memory, it is assumed to be associated with current device.
+   */
+  inline void set_backing_allocation(const shared_ptr<void> &ptr, size_t bytes,
```

Comment:
I think at this point we should also specify whether the buffer is pinned or not
```suggestion
  inline void set_backing_allocation(const shared_ptr<void> &ptr, size_t bytes, bool pinned,
```

Reply:
Done